texar/modules/classifiers/xlnet_classifier.py

```diff
@@ -0,0 +1,287 @@
+            :meth:`default_hparams` for the hyperparameter structure
+            and default values.
+    """
+
+    def __init__(self,
+                 pretrained_model_name: Optional[str] = None,
+                 cache_dir: Optional[str] = None,
+                 hparams=None):
+
+        super().__init__(hparams)
+
+        # Create the underlying encoder
+        encoder_hparams = utils.dict_fetch(hparams,
+                                           XLNetEncoder.default_hparams())
+        if encoder_hparams is not None:
+            encoder_hparams['name'] = None
+
+        self._encoder = XLNetEncoder(
+            pretrained_model_name=pretrained_model_name,
+            cache_dir=cache_dir,
+            hparams=encoder_hparams)
+
+        if self._hparams.use_projection:
+            self.projection = nn.Linear(self._encoder.output_size,
+                                        self._encoder.output_size)
+        self.dropout = nn.Dropout(self._hparams.dropout)
+
+        if self._hparams.summary_type == 'last':
+            self.summary_op = lambda output: output[:, -1]
+        elif self._hparams.summary_type == 'first':
+            self.summary_op = lambda output: output[:, 0]
+        elif self._hparams.summary_type == 'mean':
+            self.summary_op = lambda output: torch.mean(output, dim=1)
+        elif self._hparams.summary_type == 'attn':
+            raise NotImplementedError
+        else:
+            raise ValueError(
+                f"Unsupported summary type {self._hparams.summary_type}")
+
+        # Create an additional classification layer if needed
+        self.num_classes = self._hparams.num_classes
+        if self.num_classes <= 0:
+            self.hidden_to_logits = None
+        else:
+            self.hidden_to_logits = nn.Linear(self._encoder.output_size,
+                                              self._hparams.num_classes)
+
+        self.is_binary = (self.num_classes == 1) or \
+                         (self.num_classes <= 0 and
+                          self._hparams.hidden_dim == 1)
+
+    @staticmethod
+    def default_hparams() -> Dict[str, Any]:
+        r"""Returns a dictionary of hyperparameters with default values.
+
+        .. code-block:: python
+
+            {
+                # (1) Same hyperparameters as in XLNetEncoder
+                ...
+                # (2) Additional hyperparameters
+                "summary_type": "last",
+                "use_projection": True,
+                "num_classes": 2,
+                "name": "xlnet_classifier",
+            }
+
+        Here:
+
+        1. Same hyperparameters as in
+            :class:`~texar.modules.XLNetEncoder`.
+            See the :meth:`~texar.modules.XLNetEncoder.default_hparams`.
+            An instance of XLNetEncoder is created for feature extraction.
+
+        2. Additional hyperparameters:
+
+            `summary_type`: str
+                The summary type, one of:
```

Comment:
I think we could use the same. By default XLNet uses `last` summary type, which is `cls_time` in Texar (but uses state from final time step because XLNet is trained with `CLS` token at the end). Other strategies (`mean`, `attn`) were implemented in their repo but were not mentioned or used, so they probably don't perform well.

Reply:
Then let's use the previous terminology and hparam structure

Reply:
Yes, I think we here should use `clas_strategy` instead of `summary_type`.